lib/actions/apiV2.js

```diff
@@ -701,12 +742,183 @@ export const findPatternsForRoute = (params) =>
     }
   }
 
+const getWalkTime = (itin) =>
+  itin.legs.reduce(
+    (sum, leg) => (sum + leg.mode === 'WALK' ? leg.duration : 0),
+    0
+  )
+
+const aggregateAlerts = (...alerts) => alerts.flat().filter((v) => !!v)
+
+const processLeg = (leg) => ({
+  ...leg,
+  agencyBrandingUrl: leg.agency?.url,
+  agencyName: leg.agency?.name,
+  agencyUrl: leg.agency?.url,
+  alerts: aggregateAlerts(
+    leg.agency?.alerts,
+    leg.route?.alerts,
+    leg.to?.stop?.alerts,
+    leg.from?.stop?.alerts
+  ),
+  from: {
+    ...leg.from,
+    stopCode: leg.from.stop?.code,
+    stopId: leg.from.stop?.gtfsId
+  },
+  route: leg.route?.shortName,
+  routeColor: leg.route?.color,
+  routeId: leg.route?.id,
+  routeLongName: leg.route?.longName,
+  routeShortName: leg.route?.shortName,
+  routeTextColor: leg.route?.textColor,
+  to: {
+    ...leg.to,
+    stopCode: leg.to.stop?.code,
+    stopId: leg.to.stop?.gtfsId
+  },
+  tripHeadsign: leg.trip?.tripHeadsign,
+  tripId: leg.trip?.gtfsId
+})
+
+export function routingQuery(searchId = null, updateSearchInReducer) {
+  return function (dispatch, getState) {
+    const state = getState()
+    const { config, currentQuery } = state.otp
+    const persistenceMode = getPersistenceMode(config.persistence)
+    const activeItinerary = getActiveItinerary(state)
+
+    const isNewSearch = !searchId
+    if (isNewSearch) searchId = randId()
+    // Don't permit a routing query if the query is invalid
+    if (!queryIsValid(state)) {
+      console.warn('Query is invalid. Aborting routing query', currentQuery)
+      return
+    }
+
+    const { date, departArrive, routingType, time } = currentQuery
+    const arriveBy = departArrive === 'ARRIVE'
+
+    const { activeModes, modeSettings } = getActivatedModesFromQueryParams(
+      state.router.location.search,
+      state.otp.config.modes.modeButtons,
+      [
+        ...defaultModeSettings,
+        ...(state.otp.config.modes.modeSettingDefinitions || [])
+      ],
+      state.otp.config.modes.initialState
+    )
+    // Get the raw query param strings to save for the rider's search history
+    const urlSearchParams = new URLSearchParams(state.router.location.search)
+    const rawModeButtonQP = urlSearchParams.get('modeButtons')
```

Comment:
I'm wondering whether the `modeButtons` query param is needed. It mostly mirrors the `mode` query param, right?

Reply:
I'd like to remove the `mode` queryParam since that is for the old system. Mode buttons do not necessarily map one to one with modes either since a button could activate multiple modes.